core/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -300,6 +301,58 @@ public Builder addBlocks(IndexMetaData indexMetaData) {
             return this;
         }
 
+        public Builder removeIndexMetadataBlocks(IndexMetaData indexMetaData) {
```

Comment:
Could you add javadocs to the methods that you have added? I don't think I quite understand the purpose and usage of `removeIndexMetadataBlocks` and `updateIndexMetadataBlocks`. You always call both of them at the same time, which seems to be an overkill. Maybe it's still jet-leg speaking :), but the way I read the code you can call either `removeIndexMetadataBlocks` followed by `addBlocks` or just `updateIndexMetadataBlocks` alone to achieve the same effect. So, some comments on this would be helpful.